File: iOS/watf-bank/PageController/Transfer.m
#import "Util.h"
#import "Http.h"
#import "TransferResult.h"

@interface Transfer:UIViewController <UITextFieldDelegate>
@property (weak, nonatomic) IBOutlet UITextField *amount;
@property (weak, nonatomic) IBOutlet UITextField *accountNo;
@end

@implementation Transfer
- (void)viewDidLoad
{
    _amount.delegate = self;
    _accountNo.delegate = self;
}

- (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string
{
    NSCharacterSet *numbersOnly = [NSCharacterSet characterSetWithCharactersInString:@"0123456789"];
    NSCharacterSet *characterSetFromTextField = [NSCharacterSet characterSetWithCharactersInString:string];
    BOOL stringIsValid = [numbersOnly isSupersetOfSet:characterSetFromTextField];
    BOOL lengthIsValid = ([textField.text length]+[string length]>10);
    return stringIsValid && !lengthIsValid;
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    [_amount resignFirstResponder];
    [_accountNo resignFirstResponder];
}


- (IBAction)push:(id)sender {
    NSDictionary *input = [NSDictionary dictionaryWithObjectsAndKeys:[Util getKeychain:@"accountNo"],@"accountNo",_accountNo.text,@"toAccountNo",_amount.text,@"amount",[Util getKeychain:@"token"],@"token", nil];
    [Http post:@"transfer" :input :^(NSDictionary *res)
     {
         if(res==nil)
         {
             [Util alert:self :@"Connection Problem"];
             return;
         }
         if([[res[@"message"] lowercaseString] isEqualToString:@"success"])
         {
             TransferResult *t = [self.storyboard instantiateViewControllerWithIdentifier:@"TransferResult"];
             t.fromAccountV = [Util getKeychain:@"accountNo"];
             t.toAccountV = self.accountNo.text;
             t.amountV = self.amount.text;
             t.fromAccountV = [NSString stringWithFormat:@"%@ : %@",@"From",[Util getKeychain:@"accountNo"]];
             t.toAccountV = [NSString stringWithFormat:@"%@ : %@",@"To",self.accountNo.text];
             t.amountV = [NSString stringWithFormat:@"%@ : %@",@"Amount",self.amount.text];
             [self presentViewController:t animated:true completion:nil];
         }
         else
         {
             [Util alert:self :@"An Error Occurred"];
         }
     }];
}
@end

File: iOS-Swift/watf-bank/PageController/Transfer.swift
import UIKit

class Transfer : UIViewController, UITextFieldDelegate {
    
    @IBOutlet weak var accountNo: UITextField!
    @IBOutlet weak var amount: UITextField!
    
    override func viewDidLoad() {
        accountNo.delegate = self
        amount.delegate = self
    }
    
    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
        let allowChar = CharacterSet(charactersIn: "0123456789")
        return  allowChar.isSuperset(of: CharacterSet(charactersIn: string)) && (textField.text!.count+string.count)<=10
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        accountNo.resignFirstResponder()
        amount.resignFirstResponder()
    }
    
    @IBAction func push(_ sender: Any) {
        guard let token = KeyChain.load("token"), let acc = KeyChain.load("accountNo") else {
            Util.alert(self, "Invalid Token")
            return
        }
        let input = ["accountNo":acc,"toAccountNo":accountNo.text!,"amount":amount.text!,"token":token]
        Http().post(input, "transfer", completionHandler: {(re: [String:Any]) in
            var res = re as! [String:String]
            if let e = res["error"] {
                Util.alert(self, e)
            }
            else if res["message"] == "Success" {
                let t = self.storyboard!.instantiateViewController(withIdentifier: "TransferResult") as! TransferResult
                t.labelV = "Transfer Succeeded"
                t.fromAccountV = "From : "+res["username"]!
                t.toAccountV = "To : "+res["toAccount"]!
                t.amountV = "Amount : "+res["amount"]!
                self.present(t, animated: true, completion: nil)
            }
            else if let m = res["message"] {
                Util.alert(self, m)
            }
            else {
                Util.alert(self, "An Error Occurred")
            }
        })
    }
}
